python/external_config/scripts/external_runner.py

```diff
@@ -143,6 +152,44 @@ def execute_command(self):
                 self.completed.emit()
 
 
+def _handle_qt_warnings():
+    """
+    This will suppress the libpng warnings, but allow any
+    other warnings or errors from QT to print.
+    We do this because warning by default get printed to stderr
+    and so they get displayed in the browser when running actions.
+    """
+
+    def handler(*args):
+        # We handle the args this way since Qt 5 passes 3 args, type, context and message,
+        # Where as Qt 4 only passes 2, type, and message.
+        msg_type = args[0]
+        msg_string = args[-1]
+
+        # Suppress this warning.
+        if msg_string == "libpng warning: iCCP: known incorrect sRGB profile":
+            return
+
+        if msg_type in [
+            qt_importer.QtCore.QtMsgType.QtWarningMsg,
+            qt_importer.QtCore.QtMsgType.QtCriticalMsg,
+            qt_importer.QtCore.QtMsgType.QtFatalMsg,
+        ]:
+            # By default Qt would usually print these to stderr so we should do the same.
+            print(msg_string, file=sys.stderr)
```

Comment:
Not that I care that much, but we lose the context in Qt 5 with that way of printing it.

Reply:
This might be fixed by something like printing the result of args[1:] joined in a nice way.

Reply:
Hmm good point. Although if I comment out the libpng suppressing bit, it still prints that warning just the same as if I didn't take over the message handler (with QT 5):
`libpng warning: iCCP: known incorrect sRGB profile`

So for that warning at least it seems the default behaviour doesn't make use of the context.
Looking at the [example in the QT docs](https://doc.qt.io/qt-5/qtglobal.html#qInstallMessageHandler) they use the context to mention the line number. Perhaps this just doesn't make sense in a Python context where the line number would presumably be in the compiled C++ code?

Reply:
Good enough then!